x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTests.java

```diff
@@ -40,30 +40,6 @@
 
 public class AuthenticationTests extends ESTestCase {
 
-    public void testWillGetLookedUpByWhenItExists() {
```

Comment:
Optional: would cover `isFailedRunAs` here

Reply:
Good point. I adapt the tests for `isFailedRunAs`. Also add new assertions in ApiKeyServiceTests to ensure the behaviours of `getCreatorRealmName` and `getCreatorRealmType` do not change.